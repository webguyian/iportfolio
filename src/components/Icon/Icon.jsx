import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Icon = props => {
  const { name, size } = props;
  const baseClass = `ui-icon ui-icon-${name}`;
  const iconClass = `fas fa-${name}`;
  const sizeClass = size ? `fa-${size}` : false;

  return <i className={classNames(baseClass, iconClass, sizeClass)} />;
};

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  size: PropTypes.string
};

export default Icon;
