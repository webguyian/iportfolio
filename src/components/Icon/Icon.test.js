import React from 'react';
import { create } from 'react-test-renderer';

import Icon from './Icon';

describe('<Icon />', () => {
  it('renders correctly', () => {
    const component = create(<Icon name="lock" />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with size prop', () => {
    const component = create(<Icon name="lock" size="2x" />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
