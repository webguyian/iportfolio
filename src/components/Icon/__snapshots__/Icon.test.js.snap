// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders correctly 1`] = `
<i
  className="fas icon icon-lock fa-lock"
/>
`;

exports[`<Icon /> renders correctly with size prop 1`] = `
<i
  className="fas icon icon-lock fa-lock fa-2x"
/>
`;
